src/os/bluestore/BlueStore.h

```diff
@@ -324,6 +340,22 @@ class BlueStore : public ObjectStore,
       if (--nref == 0)
 	delete this;
     }
+
+    void encode(bufferlist& bl) const {
+      if (dirty) {
+	::encode(blob, blob_bl);
+	dirty = false;
+      } else {
+	assert(blob_bl.length());
+      }
+      bl.append(blob_bl);
+    }
+    void decode(bufferlist::iterator& p) {
```

Comment:
I'm wondering if we can benefit from a deferred decoding. Just store encoded blob and increment a bufferlist iterator here. And do actual decode when get_blob or dirty_blob are called for the first time. The main question here is if there are decoded but unused blobs in the current implementation.